.gitlab/package_build/dmg.yml

```diff
@@ -12,7 +12,7 @@
     - export VERSION_CACHE_CONTENT=$(cat agent-version.cache | base64 -)
     - !reference [.setup_python_mirror_linux]
     - python3 -m pip install -r tasks/libs/requirements-github.txt
-    - inv -e github.trigger-macos --workflow-type "build" --datadog-agent-ref "$CI_COMMIT_SHA" --release-version "$RELEASE_VERSION" --major-version "$AGENT_MAJOR_VERSION" --python-runtimes "$PYTHON_RUNTIMES" --destination "$OMNIBUS_PACKAGE_DIR" --version-cache "$VERSION_CACHE_CONTENT"
```

Comment:
Is it working as expected?
I feel like `INTEGRATIONS_CORE_VERSION` is not defined when the pipeline is not triggered by integrations-core.
So the command executed contains `--integrations-core-ref ""`, in this case in the invoke function `integrations_core_ref` will be `""` and not `DEFAULT_INTEGRATIONS_CORE_BRANCH`

Reply:
Maybe I am wrong but I think it works because [here](https://github.com/DataDog/datadog-agent-macos-build/blob/master/scripts/build_script.sh#L47)
the condition is false when `INTEGRATIONS_CORE_REF` is `""`

Reply:
`INTEGRATIONS_CORE_VERSION` is indeed passed as `""` and in macos build script [we look for a non-empty string](https://github.com/DataDog/datadog-agent-macos-build/blob/master/scripts/build_script.sh#L47-L49)
As a consequence we can see in the build [is getting](https://github.com/DataDog/datadog-agent-macos-build/actions/runs/8378059164/job/22941685438#step:9:448) `master`

Reply:
If it was expected it's fine. I just found it was a bit strange because we could expect that the invoke task use `DEFAULT_INTEGRATION_CORE_BRANCH` when the variable is not set in the CI. But it is not the case here.
Changing this variable has no effect on the CI, currently